### PR TITLE
Feature/cu mke50n corridor lasers

### PR DESCRIPTION
**Pull request by Jordi-Gil**

### CODE REVIEW

**Review by Yonimevicio — APPROVED** [review]

File: Penteract/Assets/Scripts/LaserTurret.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#include "Scripting/Script.h"
+
+class ComponentAnimation;
+
+class LaserTurret : public Script
+{
+	GENERATE_BODY(LaserTurret);
+
+	enum class TurretState {
+		START = 0,
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/LaserTurret.h:12]: There's no need to init enums

File: Penteract/Assets/Scripts/LaserTurret.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    switch (currentState)
+    {
+    case TurretState::IDLE_START:
+        if (coolDownOffTimer <= coolDownOff) {
+            coolDownOffTimer += Time::GetDeltaTime();
+            if (coolDownOffTimer > coolDownOff) {
+                coolDownOffTimer = 0;
+                currentState = TurretState::START;
+                animationComp->SendTrigger(states[static_cast<unsigned int>(TurretState::IDLE_START)] + states[static_cast<unsigned int>(TurretState::START)]);
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/LaserTurret.cpp:45]: You could use currentState as an int and convert the enum class into an enum so you don't need to cast everytime, but i'm just being picky here

**Review by magalenyo — APPROVED** [review]

**Review by GBMiro — APPROVED** [review]

File: Penteract/Assets/Scripts/LaserTurret.cpp (hunk 2 above)
Comment (Mefiso) [reply Penteract/Assets/Scripts/LaserTurret.cpp:45]: https://youtu.be/1taLXjVxB4U?t=71

> 
> 
> You could use currentState as an int and convert the enum class into an enum so you don't need to cast everytime, but i'm just being picky here

**Review by Mefiso — APPROVED** [review]

**Comment by jcarlos0305** [comment]
Task linked: [CU-mke50n Add lasers in the corridor](https://app.clickup.com/t/mke50n)

**Pull request merged**